File: solutions/go/grains/1/grains.go
package grains

import (
	"fmt"
	"math"
)

func Square(number int) (uint64, error) {
	if number < 1 || number > 64 {
		return 0, fmt.Errorf("invalid square %d", number)
	}
	return uint64(math.Pow(2, float64(number)-1)), nil
}

func Total() uint64 {
	var total uint64 = 0
	for i := 1; i < 65; i++ {
		if res, err := Square(i); err == nil {
			total += res
		}
	}
	return total
}

File: solutions/typescript/grains/1/grains.ts
export default class Grains {
  static square(nr: number) {
    if (nr < 1 || nr > 64) throw Error("invalid square");
    return 2 ** (nr - 1);
  }

  static total() {
    let total = 0;
    for (let i = 0; i < 64; i++) {
      total += 2 ** i;
    }
    return total;
  }
}
